File: decompiled/sources/com/hjq/widget/layout/CustomViewStub.java
package com.hjq.widget.layout;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import com.hjq.widget.R;

public final class CustomViewStub extends FrameLayout {
    private View mInflateView;
    private final int mLayoutResource;
    private OnViewStubListener mListener;

    public interface OnViewStubListener {
        void onInflate(CustomViewStub customViewStub, View view);

        void onVisibility(CustomViewStub customViewStub, int i);
    }

    public CustomViewStub(Context context) {
        this(context, (AttributeSet) null);
    }

    public CustomViewStub(Context context, AttributeSet attributeSet) {
        this(context, attributeSet, 0);
    }

    public CustomViewStub(Context context, AttributeSet attributeSet, int i) {
        this(context, attributeSet, i, 0);
    }

    public CustomViewStub(Context context, AttributeSet attributeSet, int i, int i2) {
        super(context, attributeSet, i, i2);
        TypedArray obtainStyledAttributes = context.obtainStyledAttributes(attributeSet, R.styleable.CustomViewStub);
        this.mLayoutResource = obtainStyledAttributes.getResourceId(R.styleable.CustomViewStub_android_layout, 0);
        obtainStyledAttributes.recycle();
        setVisibility(8);
    }

    public void setVisibility(int i) {
        super.setVisibility(i);
        if (this.mInflateView == null && i != 8) {
            View inflate = LayoutInflater.from(getContext()).inflate(this.mLayoutResource, this, false);
            this.mInflateView = inflate;
            FrameLayout.LayoutParams layoutParams = (FrameLayout.LayoutParams) inflate.getLayoutParams();
            layoutParams.width = getLayoutParams().width;
            layoutParams.height = getLayoutParams().height;
            if (layoutParams.gravity == -1) {
                layoutParams.gravity = 17;
            }
            this.mInflateView.setLayoutParams(layoutParams);
            addView(this.mInflateView);
            OnViewStubListener onViewStubListener = this.mListener;
            if (onViewStubListener != null) {
                onViewStubListener.onInflate(this, this.mInflateView);
            }
        }
        OnViewStubListener onViewStubListener2 = this.mListener;
        if (onViewStubListener2 != null) {
            onViewStubListener2.onVisibility(this, i);
        }
    }

    public void setCustomVisibility(int i) {
        super.setVisibility(i);
    }

    public View getInflateView() {
        return this.mInflateView;
    }

    public void setOnViewStubListener(OnViewStubListener onViewStubListener) {
        this.mListener = onViewStubListener;
    }
}

File: decompiled/sources/com/hjq/widget/layout/CustomViewStub.kt
package com.hjq.widget.layout

import android.content.Context
import android.content.res.TypedArray
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import com.hjq.widget.R

class CustomViewStub : FrameLayout {
    private var mInflateView: View? = null
    private val mLayoutResource: Int
    private var mListener: OnViewStubListener? = null

    interface OnViewStubListener {
        fun onInflate(customViewStub: CustomViewStub, view: View)

        fun onVisibility(customViewStub: CustomViewStub, visibility: Int)
    }

    constructor(context: Context) : this(context, null)

    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : this(
        context,
        attrs,
        defStyleAttr,
        0
    )

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(
        context,
        attrs,
        defStyleAttr,
        defStyleRes
    ) {
        val typedArray: TypedArray = context.obtainStyledAttributes(attrs, R.styleable.CustomViewStub)
        mLayoutResource = typedArray.getResourceId(R.styleable.CustomViewStub_android_layout, 0)
        typedArray.recycle()
        visibility = View.GONE
    }

    override fun setVisibility(visibility: Int) {
        super.setVisibility(visibility)
        if (mInflateView == null && visibility != View.GONE) {
            val inflate = LayoutInflater.from(context).inflate(mLayoutResource, this, false)
            mInflateView = inflate
            val childLayoutParams = inflate.layoutParams as FrameLayout.LayoutParams
            val parentLayoutParams = layoutParams
            childLayoutParams.width = parentLayoutParams.width
            childLayoutParams.height = parentLayoutParams.height
            if (childLayoutParams.gravity == -1) {
                childLayoutParams.gravity = 17
            }
            inflate.layoutParams = childLayoutParams
            addView(inflate)
            mListener?.onInflate(this, inflate)
        }
        mListener?.onVisibility(this, visibility)
    }

    fun setCustomVisibility(visibility: Int) {
        super.setVisibility(visibility)
    }

    fun getInflateView(): View? {
        return mInflateView
    }

    fun setOnViewStubListener(listener: OnViewStubListener?) {
        mListener = listener
    }
}
